src/Backend/Modules/Locale/Engine/LocaleAnalyser.php

```diff
@@ -47,6 +49,76 @@ private function getInBetweenStrings(string $start, string $end, string $str)
         return isset($matches[1]) ? current($matches[1]) : '';
     }
 
+    private function getDefaultModule(string $currentModuleName): string
+    {
+        return $this->application === 'Backend' ? $currentModuleName : 'Core';
+    }
+
+    private function getLocaleArrayFromRegexMatches(string $defaultModuleName, array $matches): array
+    {
+        // remove the full match
+        $matchesCount = count(array_shift($matches));
+
+        [$types, $labels, $modules] = $matches;
+
+        $locale = [];
+
+        for ($index = 0; $index < $matchesCount; ++$index) {
+            $module = empty($modules[$index]) ? $defaultModuleName : $modules[$index];
+            $locale[$types[$index]][$module][$labels[$index]] = $labels[$index];
+        }
+
+        return $locale;
+    }
+
+    private function findLocaleInJsFile(string $moduleName, string $filename, string $fileContent): array
+    {
+        $locale = [];
+        $patterns = [
+            '"\.locale\.(act|err|lbl|msg)\(\'(\w+?)\'(?:, ?\'(\w+?)\')?\)"',
+            '"\.locale\.get\(\'(\w+?)\', ?\'(\w+?)\'(?:, ?\'(\w+?)\')?\)"',
+        ];
+        foreach ($patterns as $pattern) {
+            $matches = [];
+            preg_match_all($pattern, $fileContent, $matches);
+            $locale[] = $this->getLocaleArrayFromRegexMatches($this->getDefaultModule($moduleName), $matches);
+        }
+
+        if (count($locale) === 0) {
```

Comment:
I prefer using `empty` as it increases readability and is actually better performance wise 

https://stackoverflow.com/questions/2216110/checking-for-empty-arrays-count-vs-empty

Reply:
fixed in 13b1f8df6

Reply:
fixed in 13b1f8df6